### PR TITLE
Add Caching for api calls

### PR DESCRIPTION
**Pull request by scampower3**
Caches the results from api calls for an hour. Caches Languages and ArtworkTypes for a day

### CODE REVIEW

**Review by Shadowghost — APPROVED** [review]
Maybe we want an option to reset the cache

**Review by crobibero — APPROVED** [review]

**Comment by scampower3** [comment]
@Shadowghost added a task with no default trigger to allow for purging of the cache

**Comment by BobSilent** [comment]
I would have preferred a simple button in the settings to purge all TVdB cached data only. E.g.  by adding a cancelationtoken to each entry when adding to cache.
When some in config presses the clear cache button, this token would be canceled and a new token for new entries created.

`.Compact()` tries too free up cache in general. But I understand 

> Maybe we want an option to reset the cache

As an option to resetting our cached data without the need to restart the whole application.


Also I would like to have the cache time configurable, and maybe also disabled by setting to 0.

File: Jellyfin.Plugin.Tvdb/TvdbClientManager.cs [hunk 1]
```diff
@@ -393,7 +491,30 @@ public async Task<IReadOnlyList<ArtworkType>> GetArtworkTypeAsync(CancellationTo
...
+
+    /// <summary>
+    /// Purge the cache.
+    /// </summary>
+    /// <returns>True if success else false.</returns>
+    public bool PurgeCache()
+    {
+        if (_memoryCache is MemoryCache memoryCache)
+        {
+            memoryCache.Compact(1);
```
Comment (cvium) [inline Jellyfin.Plugin.Tvdb/TvdbClientManager.cs:512]: IMemoryCache is a singleton. You're nuking the cache for the entire server.

Comment (scampower3) [reply Jellyfin.Plugin.Tvdb/TvdbClientManager.cs]: Reverted

**Comment by scampower3** [comment]
> I would have preferred a simple button in the settings to purge all TVdB cached data only. E.g. by adding a cancelationtoken to each entry when adding to cache. When some in config presses the clear cache button, this token would be canceled and a new token for new entries created.
> 
> `.Compact()` tries too free up cache in general. But I understand
> 
> > Maybe we want an option to reset the cache
> 
> As an option to resetting our cached data without the need to restart the whole application.
> 
> Also I would like to have the cache time configurable, and maybe also disabled by setting to 0.

I do have another branch that allows for time configuration for cache.

File: Jellyfin.Plugin.Tvdb/TvdbClientManager.cs (hunk 1 above)
Comment (cvium) [reply Jellyfin.Plugin.Tvdb/TvdbClientManager.cs]: You can just `new` your own memorycache if you want to keep the functionality

**Pull request merged**